staging/src/k8s.io/apimachinery/pkg/util/sets/set.go

```diff
@@ -64,6 +64,14 @@ func (s Set[T]) Delete(items ...T) Set[T] {
 	return s
 }
 
+// Clear empties the set.
+func (s Set[T]) Clear() Set[T] {
+	for key := range s {
+		delete(s, key)
```

Comment:
By the way, if `key` is NaN, this will fail. Not sure on the likelihood of this as I haven’t had a chance to familiarise myself with the repo so I could be wrong

Reply:
So it will. Float types also don’t get optimised to `runtime.mapclear`, compare https://godbolt.org/z/afTr4d6Wa and https://godbolt.org/z/ErGjb53aa

Reply:
Nice, thanks for clarifying 👍

Reply:
If a pointer to the set has been shared, the set can’t be replaced with a new one. In scenarios where the set _can_ be replaced, this doesn’t forbid that.

As far as complexity is concerned, as mentioned above the loop is typically optimised to a single call to `mapclear`.